### PR TITLE
finished quantum yield docs

### LINKED ISSUE #53: Document API: quantum yields

### PR DESCRIPTION
**Pull request by K20shores**
Link to [this branch's docs with my changes](https://ncar.github.io/tuv-x/branch/53-document-api-quantum-yields).

This should close #53. Notably missing in this PR is the description of the configuration for grids, but that is because #48 seems to be a catch-all for configuration. I will certainly add the configuration information if it is desired here, but I thought this was specifically for the code documentation.

### CODE REVIEW

File: src/quantum_yield.F90 [hunk 1]
```diff
@@ -65,14 +61,10 @@ subroutine base_constructor( this, config, grid_warehouse,                  &
...
 
-    !> Quantum yield calculator
-    class(quantum_yield_t),    intent(inout) :: this
-    !> Configuration data
-    type(config_t),            intent(inout) :: config
-    !> Grid warehouse
-    type(grid_warehouse_t),    intent(inout) :: grid_warehouse
-    !> Profile warehouse
-    type(profile_warehouse_t), intent(inout) :: profile_warehouse
+    class(quantum_yield_t), pointer :: this ! This :f:type:`~tuvx_quantum_yield/quantum_yield_t` calculator
```
Comment (mattldawson) [inline src/quantum_yield.F90:64]: does this need to be a pointer? also, the `intent(inout)` has been dropped

File: src/quantum_yield.F90 [hunk 2]
```diff
@@ -156,26 +148,22 @@ end subroutine base_constructor
...
 
-    !> Quantum yield calculator
-    class(quantum_yield_t),    intent(in)    :: this
-    !> Grid warehouse
-    type(grid_warehouse_t),    intent(inout) :: grid_warehouse
-    !> Profile warehouse
-    type(profile_warehouse_t), intent(inout) :: profile_warehouse
-    !> Calculated quantum yield
-    real(dk), allocatable                    :: quantum_yield(:,:)
+    class(quantum_yield_t), pointer :: this ! This :f:type:`~tuvx_quantum_yield/quantum_yield_t` calculator
```
Comment (mattldawson) [inline src/quantum_yield.F90:163]: does this need to be a pointer? also missing `intent(inout)`

File: src/quantum_yield.F90 (hunk 1 above)
Comment (K20shores) [reply src/quantum_yield.F90]: Fixed

File: src/quantum_yield.F90 (hunk 2 above)
Comment (K20shores) [reply src/quantum_yield.F90]: Oops, this was a copy and paste error. Fixed.

File: docs/source/api/quantum_yield.rst [hunk 3]
```diff
@@ -1,4 +1,90 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_ch2o_h2_co
+
+:math:`CH_3CHO+hv \rightarrow CH_3+HCO`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_ch3cho_ch3_hco
+
+:math:`CH_3COCH_2CH_3+hv \rightarrow CH_3CO+CH_2CH_3`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_ch3cho_ch3_hco
```
Comment (mattldawson) [inline docs/source/api/quantum_yield.rst:42]: is this the right module for this reaction? (it's the same as the one above)

File: docs/source/api/quantum_yield.rst [hunk 4]
```diff
@@ -1,4 +1,90 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_no2_tint
+
+:math:`{NO_{3}}^-_{(aq)}+hv \rightarrow {NO_2}_{(aq)}+O^-` 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_no3m_aq
+
+:math:`O_3+hv \rightarrow O_2 + O(^1D)` 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. f:automodule:: tuvx_quantum_yield_o3_o2_o3p
```
Comment (mattldawson) [inline docs/source/api/quantum_yield.rst:82]: the module should be for O1D instead of O3P

File: docs/source/api/quantum_yield.rst (hunk 3 above)
Comment (K20shores) [reply docs/source/api/quantum_yield.rst:42]: Oops. You're right

File: docs/source/api/quantum_yield.rst (hunk 4 above)
Comment (K20shores) [reply docs/source/api/quantum_yield.rst:82]: That is true.

Comment (K20shores) [reply docs/source/api/quantum_yield.rst]: fixed

File: docs/source/api/quantum_yield.rst (hunk 3 above)
Comment (K20shores) [reply docs/source/api/quantum_yield.rst]: Fixed

**Review by mattldawson — APPROVED** [review]
looks good to me!

**Pull request merged**